static/app/types/onboarding.tsx

```diff
@@ -49,7 +58,26 @@ export type OnboardingTaskDescriptor = {
       action: () => void;
       actionType: 'action';
     }
-);
+) &
+  (
+    | {
+        Component: (OnboardindgCustomComponentProps) => JSX.Element | null;
```

Comment:
Should this also be `React.ComponentType< OnboardindgCustomComponentProps>` ?

Reply:
I've decided to keep it like that to indicate that this is not a component.